src/lib.rs

```diff
@@ -539,6 +534,29 @@ fn find_placeholders(contents: &str) -> PlaceholderIter<'_> {
     PlaceholderIter(RE.captures_iter(contents))
 }
 
+fn url_to_root(source_file: &std::path::Path) -> String {
```

Comment:
Maybe `get_breadcrumbs_to_root` or simply `get_dotted_path_to_root` are better names for this function